Application/src/view/visuals/component/Message.java

```diff
@@ -233,8 +239,29 @@ public void renderDefault(GraphicsContext gc) {
 
 
         //Draw trail for the message, for each instance in the arraylist
-        for (Trail t: trails)
-            gc.drawImage(trail, t.getXcoordinate(), (t.getYcoordinate() + 18), t.getWidth(), t.getHeight());
+        for (int i = 0; i < trails.size()-1; i++){
+
+            // Index each trail
+			// Starts with the second index to fix visual issue with the first cloud sprite
+            Trail t = trails.get(i+1);
+            // Index of the last trail
+            Trail last = trails.get(trails.size()-1);
+
+            // Draw a trail from left to right
+            if (rotated) {
+                // Draw a cloud to indicate a trail
+                gc.drawImage(cloud, t.getXcoordinate(), (t.getYcoordinate() + 18), t.getWidth(), t.getHeight());
+                // Draw an arrow to indicate direction
+                gc.drawImage(rotatedTrail, last.getXcoordinate(), (last.getYcoordinate() + 18), last.getWidth(), last.getHeight());
+            }
+            // Draw a trail from right to left
+            else{
+                // Draw a cloud to indicate a trail
+                gc.drawImage(cloud, t.getXcoordinate(), (t.getYcoordinate() + 18), t.getWidth(), t.getHeight());
+                // Draw an arrow to indicate direction
+                gc.drawImage(trail, last.getXcoordinate(), (last.getYcoordinate() + 18), last.getWidth(), last.getHeight());
```

Comment:
You are just drawing the trail on top of the last trail here, please dont do that, it looks bad and its inefficient 

Adding this comment here just to show that the issue is in 2 locations